### ISSUE TITLE
issue generating transactions with bip32 methods

### ISSUE THREAD

**Issue by PeterGarner** [bitcoinjs/bitcoinjs-lib#1171]
Hello again. 
I hope this makes some kind of sense, I left the the extra code in to give you some kind of idea of what I'm trying to do.  

My problem is that on the one hand, I am unable to use the .redeem.output methods for building transactions when using BIP32 type nodes.  On the other hand, I would be happy to use the ECPair type nodes but seem to be unable to derive child keys with them (which is why I left that bit in).   
```
 const mnemonic = bip39.generateMnemonic();
 const seed = bip39.mnemonicToSeed(mnemonic);
 const master = bip32.fromSeed(seed, testnet)
 const to_txpriv = master.toBase58();   // send to db

 function getPubAddress(derived) {
     const { address } = bitcoin.payments.p2sh({
         redeem: bitcoin.payments.p2wpkh({ pubkey: derived.publicKey, network: bitcoin.networks.testnet }),
         network: bitcoin.networks.testnet
     })
     return address
 }

 const from_txpriv = bip32.fromBase58(to_txpriv, testnet) // retrieve from db
 console.log(from_txpriv)
 const keyPair32 = from_txpriv.derivePath("m/44'/0'/0'/0/0");  // **will not work with ECPair type nodes.**
 var p2sh32 = getPubAddress(keyPair32)
```
With the alternative being:
```
 // const keyPairECP = bitcoin.ECPair.makeRandom({ network: testnet })
 // console.log(keyPairECP)

 // const p2wpkh = bitcoin.payments.p2wpkh({ pubkey: keyPairECP.publicKey, network: testnet })
 // const p2sh = bitcoin.payments.p2sh({ redeem: p2wpkh, network: testnet })
 // console.log(p2sh)
```
get some btc for 2N7pF7sC7UP7LbycycpUxY8smqYDvM1pUJZ, try to send to 2My4JjiQuXKcZKkES6k2Z38ZJpxjPNXxPX9

```
 var apireturn = [{
     address: '2N7pF7sC7UP7LbycycpUxY8smqYDvM1pUJZ',
     txid: '944c2c2f5d7ae9cf7d72821f9d8e8f9aee980fc719408f1fbfc36c5b240b89f5',
     vout: 0,
     scriptPubKey: 'a9149fd06e7d188cb8825352bf30644d80c5a8fe032787',
     amount: 1.3,
     satoshis: 130000000,
     confirmations: 0,
     ts: 1533871131
 }]

  const txb = new bitcoin.TransactionBuilder(testnet)
  txb.addInput(apireturn[0].txid, apireturn[0].vout)
  txb.addOutput('2My4JjiQuXKcZKkES6k2Z38ZJpxjPNXxPX9', 129999000)
  txb.sign(0, keyPair32, p2sh32.redeem.output, null, apireturn[0].satoshis)  // **will not work with BIP32 type nodes.**

  const tx = txb.build()

  console.log(tx.toHex())
```

keyPair32 is obviously vastly different to keyPairECP but I don't know enough to find a way to use keyPair32 so that I can derive as well as transact.  Of course, I do see there is no method or function in your code for that can be called for p2sh32, I just wanted to clarify my process.  

In summary, I'm trying to derive single and multi-sig segwit addresses with the single signer being p2sh(p2wpkh) (as per #1167).  I then would like to create transactions for both.  Thanks again.

**Comment by junderw** [comment]
`p2sh32.redeem.output` should throw an error, as `p2sh32` is a string ( `var p2sh32 = getPubAddress(keyPair32)` should return address which is a String)

**Comment by junderw** [comment]
change the function to:

```javascript
function getPubAddress(derived) {
  const p2sh = bitcoin.payments.p2sh({
    redeem: bitcoin.payments.p2wpkh({ pubkey: derived.publicKey, network: testnet }),
    network: testnet
  })
  return p2sh
}
```
and it worked for me

**Comment by junderw** [comment]
If you have further questions feel free to re-open
